scio-core/src/main/scala/com/spotify/scio/coders/instances/AvroCoders.scala

```diff
@@ -153,7 +153,8 @@ trait AvroCoders {
       override def apply(writer: Schema, reader: Schema): DatumReader[T] = {
         // create the datum writer using the schema api
         // class API might be unsafe. See schemaForClass
-        val datumReader = new ReflectDatumReader[T](schemaForClass(clazz).get);
+        val schema = schemaForClass(clazz).get
+        val datumReader = new ReflectDatumReader[T](schema, schema, new ReflectData())
```

Comment:
```suggestion
        val datumReader = new ReflectDatumReader[T](writer, reader, new ReflectData())
```
And remove `setExpected` and `setSchema`